src/group_impl.h

```diff
@@ -67,6 +67,7 @@ static const secp256k1_fe secp256k1_fe_const_b = SECP256K1_FE_CONST(0, 0, 0, 0,
 static void secp256k1_ge_set_gej_zinv(secp256k1_ge *r, const secp256k1_gej *a, const secp256k1_fe *zi) {
     secp256k1_fe zi2;
     secp256k1_fe zi3;
+    VERIFY_CHECK(!a->infinity);
```

Comment:
Do we need to account for the public-facing `secp256k1_ecmult_const`, where the user can possibly pass `a = infinity` and cause this check to fail?

https://github.com/bitcoin-core/secp256k1/blob/master/src/ecmult_const_impl.h#L172
https://github.com/bitcoin-core/secp256k1/blob/master/src/ecmult_impl.h#L113
https://github.com/bitcoin-core/secp256k1/blob/master/src/ecmult_impl.h#L81

Reply:
`secp256k1_ecmult_odd_multiples_table` already has a VERIFY_CHECK for infinity, if I'm reading this correctly.

`src/modules/ecdh/main_impl.h` seems to be the only place where `secp256k1_ecmult_const` is used, and it is only used with a loaded public key, which is never infinity.

I'm somewhat skeptical that the table building function will work when given infinity.

Perhaps the documentation for `secp256k1_ecmult_const` should be updated to explicitly state that the input point cannot be infinity, and maybe a VERIFY_CHECK added there too.

Reply:
Oops, you're right, the failed check (which I triggered by calling `secp256k1_ecmult_const` with `a = infinity`) probably came from this path instead:

https://github.com/bitcoin-core/secp256k1/blob/master/src/ecmult_impl.h#L115
https://github.com/bitcoin-core/secp256k1/blob/master/src/group_impl.h#L163

It looks like all other calls (in `secp256k1_ecmult_odd_multiples_table` and `secp256k1_ge_set_all_gej_var`) are protected by a check `!a.infinity`.

Reply:
Adding a comment and a VERIFY_CHECK to `secp256k1_ecmult_const` sounds good to me, I will ACK after that.

Reply:
Done.